### PR TITLE
2021 05 04 bug chaining

### PR DESCRIPTION
**Pull request by tomrittervg**
This isn't done yet, I need to add more tests and test the db migration code, but I think it's done enough to request a review

### CODE REVIEW

**Comment by tomrittervg** [comment]
Okay, I need to add more tests to this, but at least the code is correct I think.

**Comment by tomrittervg** [comment]
- [X] Test the database migration code on prod data
- [ ] Test where we create a commit alert; bump ff versions, and then mark the new one as affected.
- [ ] Test where we create 2 commit alerts; bump ff versions, and then mark them as affected. Verify the comment in the second includes the first.
- [ ] Test where we create one commit alert, close it, create another, confirm the comment doesn't refernce it, bump ff versions, and then mark the second one only as affected.
- [ ] Test where we create a commit alert, bump ff versions, and the old one is unaffected.
- [ ] Test where we create a commit alert, bump ff versions, mark it as affected, someone marks it as wontfix, and we don't re-edit it.
- [ ] Test where we create a vendoring job, and while it's still running we create a new one.
- [ ] Test where we create a vendoring job, it completes, and we create a new one.
- [ ] Test where we create a vendoring job, and while it's still running we create a new one, and while it's still running we create a new one.
- [ ] Test where we create a vendoring job, it completes, and we create a new one, it completes, and we create a new one.
- [ ] Test where we create a vendoring job, it completes. Then we bump FF versions and mark the old job as still affected.
- [ ] Test where we create a vendoring job, it completes. Then we bump FF versions and mark the old job as still affected. Then someone in bugzilla marks it as 'wontfix' and we _don't_ re-edit it

File: tests/functionality_commitalert.py [hunk 1]
```diff
@@ -112,6 +112,10 @@ def __init__(self, config):
         self._expected_commits_seen_func = config['expected_commits_seen_func']
         self._get_filed_bug_id_func = config['get_filed_bug_id_func']
         self._filed_bug_ids_func = config['filed_bug_ids_func']
+        if config['assert_affected_func']:
```
Comment (tomrittervg) [inline tests/functionality_commitalert.py:115]: Add AssertFalse to this test

File: tests/functionality_all_platforms.py [hunk 2]
```diff
@@ -370,12 +376,22 @@ def testExistingJobClassifiedFailures(self):
     # Create -> Jobs are Running -> Build Failed
     @logEntryExit
     def testExistingJobBuildFailed(self):
+        global was_abandoned
+        was_abandoned = False
+
+        def abandon_callback(cmd):
+            global was_abandoned
+            was_abandoned = True
```
Comment (tomrittervg) [inline tests/functionality_all_platforms.py:386]: Not checking this variable

**Comment by tomrittervg** [comment]
Okay; I resolved the issues I noticed while doing the code review

File: components/scmprovider.py [hunk 3]
```diff
@@ -121,10 +133,9 @@ def check_for_update(self, library, task, ignore_commits_from_these_jobs):
...
             if not all_new_upstream_commits:
                 self.logger.log("Checking for updates to %s but no new upstream commits were found from our current in-tree revision %s." % (library.name, library.revision), level=LogLevel.Info)
-                return []
+                return [], []
 
             # We do have new upstream commits.
-            # Step 3: Get the most recent job for the library
             most_recent_job = ignore_commits_from_these_jobs[0] if ignore_commits_from_these_jobs else None
 
             # Step 4: Check if the most recent job was performed on a revision _after_ the library's current revision or _before_.
```
Comment (metalcanine) [inline components/scmprovider.py:141]: Steps in comments need to be renumbered

File: tasktypes/vendoring.py [hunk 4]
```diff
@@ -72,30 +75,75 @@ def process_task(self, library, task):
 
     @logEntryExit
     def _process_new_job(self, library, task, new_version, timestamp):
-        see_also = []
+        # We need this really dumb _styleN functions because otherwise flake8 will complcain about
```
Comment (metalcanine) [inline tasktypes/vendoring.py:78]: complcain -> complain

**Review by metalcanine — APPROVED** [review]
just a couple of nits in comments.
looks amazing! thanks!

File: components/scmprovider.py (hunk 3 above)
Comment (tomrittervg) [reply components/scmprovider.py:141]: This is also a bug in the main branch I'm going to fix this in a separate commit at the end so i don't need to rebase.

**Pull request merged**